From wfp5p@virginia.edu  Mon Jul 27 14:07:55 2009
From: Bill Pemberton <wfp5p@virginia.edu>
Date: Mon, 27 Jul 2009 16:47:29 -0400
Subject: Staging: hv: fix blkvsc_ioctl() parameters
To: greg@kroah.com
Cc: hjanssen@microsoft.com
Message-ID: <1248727665-23969-8-git-send-email-wfp5p@virginia.edu>


blkvsc_ioctl() had the wrong parameter list for  struct
block_device_operations

Signed-off-by: Bill Pemberton <wfp5p@virginia.edu>
Cc: Hank Janssen <hjanssen@microsoft.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 drivers/staging/hv/blkvsc_drv.c |    8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

--- a/drivers/staging/hv/blkvsc_drv.c
+++ b/drivers/staging/hv/blkvsc_drv.c
@@ -129,8 +129,8 @@ static int blkvsc_release(struct gendisk
 static int blkvsc_media_changed(struct gendisk *gd);
 static int blkvsc_revalidate_disk(struct gendisk *gd);
 static int blkvsc_getgeo(struct block_device *bd, struct hd_geometry *hg);
-static int blkvsc_ioctl(struct inode *inode, struct file *filep, unsigned cmd, unsigned long arg);
-
+static int blkvsc_ioctl(struct block_device *bd, fmode_t mode,
+			unsigned cmd, unsigned long argument);
 static void blkvsc_request(struct request_queue *queue);
 static void blkvsc_request_completion(STORVSC_REQUEST* request);
 static int blkvsc_do_request(struct block_device_context *blkdev, struct request *req);
@@ -1451,9 +1451,9 @@ int blkvsc_getgeo(struct block_device *b
     return 0;
 }
 
-static int blkvsc_ioctl(struct inode *inode, struct file *filep, unsigned cmd, unsigned long arg)
+static int blkvsc_ioctl(struct block_device *bd, fmode_t mode,
+			unsigned cmd, unsigned long argument)
 {
-	struct block_device *bd = inode->i_bdev;
 	struct block_device_context *blkdev = bd->bd_disk->private_data;
 	int ret=0;
 
